client/me/purchases/purchases-mixin.js

```diff
@@ -6,9 +6,45 @@ import page from 'page';
 /**
  * Internal Dependencies
  */
+import analytics from 'analytics';
 import paths from './paths';
 
 export default {
+	componentWillMount() {
+		if ( this.isDataLoading() ) {
+			return;
+		}
+
+		const purchase = this.getPurchase();
+		this.recordPageView( purchase );
+	},
+
+	componentWillReceiveProps( nextProps ) {
+		// Return if the purchases haven't loaded yet
+		if ( ! nextProps.selectedPurchase.hasLoadedFromServer ) {
+			return;
+		}
+
+		// Return if the purchases have already been loaded into the props
+		if ( this.props.selectedPurchase.data && nextProps.selectedPurchase.data.id === this.props.selectedPurchase.data.id ) {
```

Comment:
I believe one of the reasons we added `hasLoadedFromServer` was so that we don't have to check for the presence of `data` anymore. If I'm not mistaken, we can combine the two `if` blocks here:

```js
if ( ! this.props.selectedPurchase.hasLoadedFromServer ||
  ! nextProps.selectedPurchase.hasLoadedFromServer ||
  nextProps.selectedPurchase.data.id === this.getPurchase().id ) {
  return null;
}
```

This also exposes another benefit of converting this mixin to a series of pure functions, as this could just be:

```js
if ( isDataLoading( this.props ) ||
  isDataLoading( nextProps ) ||
  getPurchase( this.props ).id === getPurchase( nextProps ).id ) {
  return null;
}
```